pkg/validation/spec/swagger.go

```diff
@@ -57,6 +65,25 @@ func (s *Swagger) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+func (s *Swagger) UnmarshalNextJSON(opts jsonv2.UnmarshalOptions, dec *jsonv2.Decoder) error {
+	var x struct {
+		*Extensions
+		*SwaggerProps
```

Comment:
why do some of these embed pointers and others don't? also, isn't Extensions already a pointer type, since it's a `map[string]interface{}`?

Reply:
Ah, this was a change of mine to avoid the struct copy at the end of the other functions by unmarshalling directly into `s` struct. It works, but did not appreciably improve performance or allocations. I'll change it back to be less confusing and more consistent.